syntax = "proto3";

package latte;

// Specifies the shape (dimensions) of a Blob.
message BlobShape {
  repeated int64 dim = 1;
}

message BlobProto {
  BlobShape shape = 1;
  repeated float data = 2;
  repeated float diff = 3;
  repeated double double_data = 4;
  repeated double double_diff = 5;
}

// The BlobProtoVector is simply a way to pass multiple blobproto instances
// around.
message BlobProtoVector {
  repeated BlobProto blobs = 1;
}

message Datum {
  int32 channels = 1;
  int32 height = 2;
  int32 width = 3;
  // the actual image data, in bytes
  bytes data = 4;
  int32 label = 5;
  // Optionally, the datum could also hold float data.
  repeated float float_data = 6;
}